Documentation/internal/README.md

```diff
@@ -1,5 +1,160 @@
 # Internal Documentation
 
-TODO(derekparker)
+_This directory holds documentation around the internals of the debugger and how it works._
 
-This directory will hold documentation around the internals of the debugger and how it works.
+This is a general overview of the `delve` internal components.
+
+`delve` consists of the debugger engine which can read and understand Go process memory (located
+in package `proc`), [dwarf](http://dwarfstd.org/doc/dwarf-2.0.0.pdf) 
+[parser](https://golang.org/pkg/debug/dwarf/) and wrapper (go specifics), `terminal` helper,
+`service` set of the RPC servers with user friendly `service/debugger` API.
+
+# `proc` package
+
+The `proc` package is the heart of the `delve` program. It implements low-level debugging interface,
+remote process and memory manipulation, disasm and etc. `proc` is based on the `ptrace(2)` syscall
+to control remote processes and `dwarf` to read debug information from the debugging process.
+
+`proc` consists of:
+
+- `Arch` representing CPU architecture
+- Breakpoints management
+- `Disassemble` disassembles target memory between two points into sequence of `AsmInstructions`,
+  with the help of `rsc.io/x86/x86asm`
+- `Eval` evaluator of the Go-like expressions (go/ast, go/constant, go/parser, go/printer, go/token)
+- `GoVersion` target compiler parser
+- `memoryReadWriter` process memory IO
+- `loadModuleData` go module parser
+- `Process` describes debugging process with mid-level debug API (handling of the remote processes, 
+  de/attach, start/stop/stepping, locations, whatever lister, breakpoints managing or return info).
+- `ptrace` [syscall](http://tldp.org/LDP/LG/issue81/sandeep.html) go binding used to do low level 
+  control of the remote processes, memory IO, start/stop/stepping execution and etc.
+- `Registers` interface
+- `Stack` iterator over stack frames
+- `Thread` management
+- `Types` parser
+- `Variables` parser
+
+Cross platform debug support is implemented for Linux, Darwin (MacOSX) and Windows. 
+Windows lacks support of process attach for now.
```

Comment:
You should be able to attach to a running Windows process if you use current tip. See f09ef23f9ec650f18f2530d7164229ea4a119700 for details.